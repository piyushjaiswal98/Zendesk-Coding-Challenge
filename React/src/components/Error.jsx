import React, { useEffect, useState } from 'react';
import './Error.scss';

const Error = (props) => {

    const [data, setData] = useState();

    useEffect(() => {
        setData(props.error);
    },[props.error]);

    return(

        <div className="error" data-testid="Error-1">
            <p>Error: {data}</p>
        </div>
    )

}

export default Error;

