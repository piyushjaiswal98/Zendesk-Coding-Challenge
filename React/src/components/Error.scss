.error{
    width: 99%;
    height: 80px;
    margin-bottom: 10px;
    background-color: white;
    border-radius: 5px;
    color: red;
    padding-top: 15px;
    p{
        margin: 0 auto;
    }
}