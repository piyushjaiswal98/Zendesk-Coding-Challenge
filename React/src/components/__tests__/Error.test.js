import { render, screen, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Error from '../Error';

afterEach(()=> {
    cleanup();
})

test('Should render Error Component', () => {
    render(<Error error={"Couldn't Authenticate you"}/>);
    const errorElement = screen.getByTestId('Error-1');
    expect(errorElement).toBeInTheDocument();
})

test('Matches Snapshot', () => {
    const tree = renderer.create(<Error error={"Couldn't Authenticate you"}/>).toJSON();
    expect(tree).toMatchSnapshot();
})