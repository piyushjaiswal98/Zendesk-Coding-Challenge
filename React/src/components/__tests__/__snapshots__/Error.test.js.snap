// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches Snapshot 1`] = `
<div
  className="error"
  data-testid="Error-1"
>
  <p>
    Error: 
  </p>
</div>
`;
